src/muqsit/pmarmorstand/entity/ArmorStandEntity.php

```diff
@@ -128,12 +128,16 @@ public function saveNBT() : CompoundTag{
 		$nbt = parent::saveNBT();
 
 		$armor_pieces = [];
-		foreach($this->getArmorInventory()->getContents() as $slot => $item){
-			$armor_pieces[] = $item->nbtSerialize($slot);
+		foreach($this->getArmorInventory()->getContents() as $slot => $item) {
+			if (!$item->isNull()) {
+				$armor_pieces[] = $item->nbtSerialize($slot);
+			}
 		}
 		$nbt->setTag(self::TAG_ARMOR_INVENTORY, new ListTag($armor_pieces, NBT::TAG_Compound));
 
-		$nbt->setTag(self::TAG_HELD_ITEM, $this->item_in_hand->nbtSerialize());
+		if (!$this->item_in_hand->isNull()) {
```

Comment:
Formatting correction: if statement shouldn't have surrounding spaces (its the same formatting that PocketMine uses)